### ISSUE TITLE
add a PiP mode to the Android app

### ISSUE THREAD

**Issue by bk138** [bk138/multivnc#31]
If you'd like to put out an incentive for implementing this feature, you can do so at https://issuehunt.io/r/bk138/multivnc

**Is your feature request about the Desktop Multivnc or the Mobile MultiVNC?**

* [ ] Desktop (wxWidgets)
* [x] Mobile (Android)

**Is your feature request related to a problem? Please describe.**

A picture-in-picture mode would allow keeping an eye on the session while doing something else.

**Describe the solution you'd like**

* https://developer.android.com/guide/topics/ui/picture-in-picture has guidelines

**Comment by gujjwal00** [comment]
I am planning to work on this. Here is the behavior I am thinking about:

- Enabled by default on supported devices
- On `onUserLeaveHint`, if connected, enter PiP mode
- Set PiP window's aspect ratio to match with framebuffer
- After entering
   - Set zoom level to fill the entire `VncCanvas`
   - Hide controls
- After exiting 
   - Set zoom to 100% or remember the old zoom?
   - Show controls
- Auto-close PiP if disconnected?
- If we enter PiP mode and user starts a new connection by going back to app
   - We can start a new `VncCanvasActivity`, leaving the PiP activity as is, *or*
   - We declare `VncCanvasActivity` as `singleTask`. We will then receive `onNewIntent` call where we can start a new fullscreen `VncCanvasActivity` and close the one in PiP mode.

**Comment by bk138** [comment]
Super nice @gujjwal00! You probably already know https://github.com/bk138/multivnc/tree/feat-android-pip where some work has started already. Beforehand, it's probably good to reason about the actual benefits of this. I started working on this as some sort of "fancy" feature but then realised it's of not much use at least for me. Maybe some kinda sysadmin use case? I'm pretty sure that if that feature comes, it should be configurable.

**Comment by gujjwal00** [comment]
Yes I was planning to pickup from https://github.com/bk138/multivnc/tree/feat-android-pip .

I personally don't have much use either (just once in a while).

So probably defer it until concrete use cases arise?

**Comment by bk138** [comment]
> Yes I was planning to pickup from https://github.com/bk138/multivnc/tree/feat-android-pip .
> 
> I personally don't have much use either (just once in a while).
> 
> So probably defer it until concrete use cases arise?

OK with me, it's rather a nice-to-have. Maybe someone chimes in here and expresses dire need, than it can be elevated :-)
